IdentityCore/src/cache/mac/MSIDMacTokenCache.m

```diff
@@ -433,7 +433,15 @@ - (BOOL)setItemImpl:(MSIDTokenCacheItem *)item
     // Copy the item to make sure it doesn't change under us.
     item = [item copy];
     
-    if (!key.service || !key.account)
+    NSString *account = key.account;
+    
+    if (!account)
+    {
+        // If we don't have one (ADFS case) then use an empty string
+        account = @"";
+    }
+    
+    if (!key.service || !account)
```

Comment:
the later part `!account` is unnecessary now? :)

Reply:
Fixed. Ideally, I don't want to have the account fallback to an empty string, but it turned out it's necessary for backward compatibility with ADAL...